doc/api/cli.md

```diff
@@ -194,6 +194,21 @@ added: v0.10
 Aborting instead of exiting causes a core file to be generated for post-mortem
 analysis using a debugger (such as `lldb`, `gdb`, and `mdb`).
 
+### `--abort-on-unhandled-rejection`
+<!-- YAML
+added: REPLACEME
+-->
+
+Aborting instead of exiting causes a core file to be generated for post-mortem
+analysis using a debugger (such as `lldb`, `gdb`, and `mdb`).
+
+This option only affects `Promise` rejections. It is only available on POSIX
+systems. It is implemented by keeping `fork()`ed copies of the process alive
+until it is known whether the `Promise` is handled within the same event loop
+iteration or is left unhandled.
+This makes unhandled rejections a significantly more complex operation,
```

Comment:
As scary as this already sounds, I think it remains a massive understatement.  Fork, particularly at high frequency, has some pretty significant performance impact:

- fork itself can take a bunch of time
- after the fork completes, performance is impacted in both parent and child processes as subsequent writes require additional page faults, even if the memory is already resident
- on systems that avoid overcommitting swap (including illumos and Linux, when configured to do so), the memory used by the child process is counted against the swap cap.  In cloud environments with tuned caps, this can quickly exhaust swap, causing lots of other operations to fail.
- If the child ends up writing to the shared pages, then the total memory used by the parent and child goes up, which can result in paging (again, particularly in environments with memory caps configured) that can dramatically reduce performance.

I think it's good that the documentation mentions the use of `fork`, and I think it would benefit from a firmer message about the possible impact.

Reply:
>  Fork, particularly at high frequency, has some pretty significant performance impact:

Yes, I am aware that that makes this option unsuitable depending on your application type. I can try to incorporate some of your points into the doc.

> If the child ends up writing to the shared pages, then the total memory used by the parent and child goes up, which can result in paging (again, particularly in environments with memory caps configured) that can dramatically reduce performance.

Why would this happen (for more than one or two pages of the stack)? The child process really isn’t doing much here.

Reply:
Any pages marked copy-on-write by fork() get cloned when the parent touches them.  If the parent does (for example) a compacting GC, that's a significant number of pages.

Reply:
Is there any way we can limit GC or at least discourage it while the forked process lives?

Reply:
It may be obvious but I wonder if reminding people they should ensure ulimit is set to allow core dumps would be helpful.